system/include/emscripten/val.h

```diff
@@ -664,8 +665,8 @@ struct BindingType<T, typename std::enable_if<std::is_base_of<val, T>::value &&
     _emval_incref(handle);
     return handle;
   }
-  static val fromWireType(WireType v) {
-    return val::take_ownership(v);
+  static T fromWireType(WireType v) {
+    return T(val::take_ownership(v));
```

Comment:
I'm not sure I understand why this is needed

Reply:
For this example let's say `<T>` is `DerivedVal`:
Previously the conversion constructor was automatically being called to go from `val` to a `DerivedVal`. Now that the `DerivedVal` constructor is explicit you must explicitly construct a `DerivedVal` with a `val`.

Reply:
Is that related to the namespace issue though?   Was there a clang-tidy warning for this?

Reply:
Thanks!